yt_dlp/extractor/smotrettv.py

```diff
@@ -0,0 +1,139 @@
+
+        video_page_url = urllib.parse.urljoin(base_url(url), extract_attributes(
+            get_element_by_class('video-content', webpage))['src'])
+        video_page = self._download_webpage(
+            video_page_url, video_id, impersonate=True,
+            note='Downloading video page', errnote='Unable to download video page')
+
+        if m3u8_urls := self._search_json(
+                r'var\s+streams\s*=', video_page, 'stream urls', video_id,
+                contains_pattern=r'(\[.+\])', default=[]):
+            formats, subtitles = [], {}
+            for m3u8_url in traverse_obj(m3u8_urls, (..., {url_or_none})):  # filter out "/error.m3u8"
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                    m3u8_url, video_id, m3u8_id=get_domain(m3u8_url), ext='mp4')
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
+            return {
+                'id': video_id,
+                'title': title,
+                'description': self._og_search_description(webpage),
+                'thumbnail': self._og_search_thumbnail(webpage),
+                'formats': formats,
+                'subtitles': subtitles,
+                'is_live': True,
+            }
+
+        if player_url := self._html_search_regex(
+                r'<iframe src="([^"]+)"', video_page, 'embedded player url', default=None):
+            return self.url_result(player_url, None, video_id, title, url_transparent=True)
+
+
+class SmotretTVLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?smotret-tv\.live/(?P<id>[^/#?.]+)'
+
+    _TESTS = [{
+        'url': 'http://smotret-tv.live/8-kanal.html',
+        'info_dict': {
+            'id': '8-kanal',
+            'ext': 'mp4',
+            'title': 're:8 канал смотреть онлайн бесплатно прямой эфир [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
+            'description': str,
+            'thumbnail': 'http://smotret-tv.live/images/8-kanal.png',
+            'live_status': 'is_live',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'http://smotret-tv.live/zvezda',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        base = base_url(url)
+
+        logo_url = None
+        if postopis := get_element_by_class('postopis', webpage):
+            if logo_path := self._search_regex(
+                    r'<img\s+src\s*=\s*"(?P<path>[^"]+)".+?/>', postopis, 'logo path', group='path', default=None):
+                logo_url = urllib.parse.urljoin(base, logo_path)
+
+        player = self._download_webpage(f'http://cdntvmedia.com/{video_id}.php', video_id, headers={'Referer': base})
+        m3u8_urls = self._search_regex(
+            r'Playerjs\s*\(\s*{.+?\s*file:\s*(["\'])(?P<url>https://[^"\']+)\1', player, 'player', group='url')
```

Comment:
Or:

```suggestion
        m3u8_urls = self._search_regex(
            r'Playerjs\s*\(\s*{.+?\s*file:\s*(["\'])(?P<url>https://(?:(?!\1).)+)\1', player, 'player', group='url')
```

To myself: do not forget what I learned.